core/src/main/clojure/xtdb/expression.clj

```diff
@@ -1500,6 +1504,7 @@
           (assert return-type (pr-str expr))
           {:!projection-fn (delay
                              (-> `(fn [~(-> rel-sym (with-tag RelationReader))
+                                       ~(-> schema-sym (with-tag IPersistentMap))
```

Comment:
Having the Kotlin interfaces define this as a Java `Map` but tag it as a Clojure `IPersistentMap` felt correct here. But wanted to flag.